### DISCUSSION TITLE
post/2024/authenticate-with-apim-managed-identity-to-your-backend-services/

### DISCUSSION THREAD

**Discussion by giscus** [Jandev/janvnl-comments#2] [Announcements]
# post/2024/authenticate-with-apim-managed-identity-to-your-backend-services/

Aside from Azure Traffic Manager, Azure Functions, and Azure Service Bus, Azure API Management (APIM) is one of my favourite services to use in just about any solution.
A useful little nugget for APIM is it’s able to have its own Managed Identity. You can choose to use a System Managed Identity or a User Managed Identity. Both options have pros and cons.
When you have configured APIM with a managed identity, this identity can be used to authenticate with the backend services.

https://jan-v.nl/post/2024/authenticate-with-apim-managed-identity-to-your-backend-services/

**Comment by simonknutzen88** [comment]
Thanks a lot for the hint to use managed identity of APIM to authenticate with the backend.
I'm using this method to authenticate a API request against a azure web app as the backend. The token is generated and sent, but I get a "403 Forbidden" 
I think the managed ID of the APIM does not have permissin to access the web app API and I don't know how I can grant the correct rights to it. 
Do you have an idea how I could manage that?

Cheers,
Simon

**Reply by simonknutzen88** [reply]
Never mind, I solved it. I had to add the enterprise application of my APIM to the list of allowed apps in my identity provider config.

**Reply by Jandev** [reply]
Great you were able to solve it!

> I had to add the enterprise application of my APIM to the list of allowed apps 

Based on this, I think you are using Scopes to authorize identities in your backend. When adding the identity to the Allowed Apps you also had to specify a scope (or multiple) to be added to this Allowed App.
What's happening now, every request you make will have this scope/scopes added to the token. Depending on your use case this can be fine. However, this line of the post still applies:

> By using this feature, every request to the backend will use the token of the Managed Identity and not of your users or services making authenticated requests to APIM.

So do make sure this is something you want. Otherwise,  implementing an on-behalf-of flow might be more useful. I can't find the MS Learn page, but here's the raw document on GitHub on setting up APIM + OBO flow: https://github.com/swgriffith/azure-guides/blob/master/aad-oauth2-on-behalf-of.md

**Reply by simonknutzen88** [reply]
Thanks a lot for the suggestion. I will bookmark the OBO Workflow for later use, but for my current use case it is fine to use the token of the managed ID for all calls.
